plugin/pulpcore/plugin/changeset/iterator.py

```diff
@@ -166,13 +169,21 @@ def _iter(self):
         Yields:
             Download: The appropriate download object.
         """
-        for artifact in ArtifactIterator(self.content):
-            if self.deferred or artifact.model.downloaded:
+        for remote_artifact in ArtifactIterator(self.content):
+            q = Q()
+            for algorithm in hashlib.algorithms_guaranteed:
+                if getattr(remote_artifact.model, algorithm):
+                    kwargs = {algorithm: getattr(remote_artifact.model, algorithm)}
+                    q |= Q(**kwargs)
```

Comment:
Django already has a convention for getting the natural key for a model.  The `natural_key()` method normally returns a tuple which is obviously not sufficient for this.  But, I think we should follow the same pattern and have a method on `RemoteArtifact` that can be used to fetch it from the DB.  Instead of here.